### PR TITLE
feat: add publish-external command to slash commands to publish external images

### LINKED ISSUE #8138: Create `/publish` command to external images
## Tell us about the problem you're trying to solve
Right now we don't have the option to release new connectors outside the main repository.
The main reason is because we use a GCS cache to get specs.
We need to create another `/publish` command works with external images, example is https://github.com/airbytehq/airbyte/pull/7837

## Describe the solution you’d like
A clear and concise description of what you want to see happen, or the change you would like to see

## Describe the alternative you’ve considered or used
A clear and concise description of any alternative solutions or features you've considered or are using today.

## Are you willing to submit a PR?

Remove this with your answer :-)

### PR DESCRIPTION
**Pull request by harshithmullapudi**
## What
Closes https://github.com/airbytehq/airbyte/issues/8138

### CODE REVIEW

File: .github/workflows/publish-external-command.yml [hunk 1]
```diff
@@ -0,0 +1,135 @@
+name: Publish External Connector Image
+on:
+  workflow_dispatch:
+    inputs:
+      connector:
+        description: "Airbyte Connector"
+        required: true
+      comment-id:
+        description: "The comment-id of the slash command. Used to update the comment with the status."
+        required: false
```
Comment (marcosmarxm) [inline .github/workflows/publish-external-command.yml:10]: You need to add a new input rigth?
`/publish-external harshith/source-api 0.1.0`

File: .github/workflows/publish-external-command.yml [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+            > :clock2: ${{github.event.inputs.connector}} https://github.com/${{github.repository}}/actions/runs/${{github.run_id}}
+      - name: Checkout Airbyte
+        uses: actions/checkout@v2
+        with:
+          repository: ${{github.event.pull_request.head.repo.full_name}} # always use the branch's repository
+      - name: Install Unzip for Databricks
+        if: github.event.inputs.connector == 'connectors/destination-databricks'
+        run: |
+          apt-get update && apt-get install -y unzip
+      - name: Install Java
```
Comment (marcosmarxm) [inline .github/workflows/publish-external-command.yml:70]: This part is not necessary I think... we're going only to run the spec cache service and after update the seed file.

File: .github/workflows/publish-external-command.yml [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+      - name: Install Pyenv and Tox
+        # Beside PyEnv, this does not set any runtimes up because it uses an AMI image that has everything pre-installed. See https://github.com/airbytehq/airbyte/issues/4559/
+        run: |
+          python3 -m pip install --quiet virtualenv==16.7.9 --user
+          python3 -m virtualenv venv
+          source venv/bin/activate
+          pip install --quiet tox==3.24.4
+      - run: |
+          echo "$SPEC_CACHE_SERVICE_ACCOUNT_KEY" > spec_cache_key_file.json && docker login -u airbytebot -p ${DOCKER_PASSWORD}
+          ./tools/integrations/manage.sh publish_external airbyte-integrations/${{ github.event.inputs.connector }} ${{ github.event.inputs.run-tests }} --publish_spec_to_cache
```
Comment (marcosmarxm) [inline .github/workflows/publish-external-command.yml:83]: You need to change here as your script works.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: .github/workflows/publish-external-command.yml [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+      - name: Search for valid connector name format
+        id: regex
+        uses: AsasInnab/regex-action@v1
+        with:
+          regex_pattern: "^(connectors|bases)/[a-zA-Z0-9-_]+$"
+          regex_flags: "i" # required to be set for this plugin
+          search_string: ${{ github.event.inputs.connector }}
+      - name: Validate input workflow format
+        if: steps.regex.outputs.first_match != github.event.inputs.connector
+        run: echo "The connector provided has an invalid format!" && exit 1
```
Comment (marcosmarxm) [inline .github/workflows/publish-external-command.yml]: Same here, maybe add a logic to regex the pattern: `user/source-image` `0.1.0`?

File: .github/workflows/publish-external-command.yml [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+          body: |
+            > :clock2: ${{github.event.inputs.connector}} https://github.com/${{github.repository}}/actions/runs/${{github.run_id}}
+      - name: Checkout Airbyte
+        uses: actions/checkout@v2
+        with:
+          repository: ${{github.event.pull_request.head.repo.full_name}} # always use the branch's repository
+      - name: Install Unzip for Databricks
+        if: github.event.inputs.connector == 'connectors/destination-databricks'
+        run: |
+          apt-get update && apt-get install -y unzip
```
Comment (marcosmarxm) [inline .github/workflows/publish-external-command.yml]: Can remove this.

File: .github/workflows/publish-external-command.yml [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+      - name: Install Pyenv and Tox
+        # Beside PyEnv, this does not set any runtimes up because it uses an AMI image that has everything pre-installed. See https://github.com/airbytehq/airbyte/issues/4559/
+        run: |
+          python3 -m pip install --quiet virtualenv==16.7.9 --user
+          python3 -m virtualenv venv
+          source venv/bin/activate
+          pip install --quiet tox==3.24.4
```
Comment (marcosmarxm) [inline .github/workflows/publish-external-command.yml:65]: Not needed to, right?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

**Comment by marcosmarxm** [comment]
> /publish-external marcosmarxm/source-test 0.1.0

not sure if we need to merge first :p

File: .github/workflows/publish-external-command.yml (hunk 5 above)
Comment (harshithmullapudi) [reply .github/workflows/publish-external-command.yml]: I removed this

File: .github/workflows/publish-external-command.yml (hunk 6 above)
Comment (harshithmullapudi) [reply .github/workflows/publish-external-command.yml:65]: I am not sure of the usage and anyways it a small installation so left it?

**Comment by harshithmullapudi** [comment]
I think we have to merge to test it 🙄 😃

**Comment by harshithmullapudi** [comment]
@marcosmarxm I have tried the command in this repo https://github.com/airbytehq/github-workflow-test-repo-base. Command in itself is working

**Review by marcosmarxm — APPROVED** [review]
I think it's ready to go, I'd recommend to merge and try to apply one test image.

**Comment by harshithmullapudi** [comment]
@marcosmarxm any suggestions on test image ?

**Pull request merged**